File: 정수빈/Programmers/Level3/네트워크.py
# https://school.programmers.co.kr/learn/courses/30/lessons/43162
# 설명: dfs로 푸는 문제
# 해결 못함 ㅠ

def solution(n, computers):
    
    visited = [False for _ in range(n)]
    count = 0
    def dfs(v):
        visited[v] = True
        
        for neig in range(n):
            if not visited[neig] and computers[v][neig] == 1:
                dfs(neig)
    
    for i in range(n):
        if not visited[i]:
            dfs(i)
            count +=1
    
    return count

File: 장세훈/Programmers/Level3/네트워크.java
class 네트워크 {
    /*
         https://school.programmers.co.kr/learn/courses/30/lessons/43162
    */

    int answer = 0;
    boolean[] visit;

    public int solution(int n, int[][] computers) {
        visit = new boolean[computers.length];

        for (int i = 0; i < computers.length; i++) {
            if (!visit[i]) {
                dfs(i, computers);
                answer++;
            }
        }

        return answer;
    }

    void dfs(int index, int[][] computers) {
        visit[index] = true;

        for (int i = 0; i < computers[index].length; i++) {
            if (computers[index][i] == 1 && !visit[i]) {
                dfs(i, computers);
            }
        }
    }
}